File: src/Domain/Library/Publication.cs
using SharedKernel;

namespace Domain.Library;

public sealed class Publication : Entity
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public int TypeId { get; set; }
    public string Concessionaire { get; set; } = string.Empty;
    public string ResponsibleDirector { get; set; } = string.Empty;
    public string Editor { get; set; } = string.Empty;
    public string Year { get; set; } = string.Empty;
    public string JournalNo { get; set; } = string.Empty;
    public string PublishDate { get; set; } = string.Empty;
    public int PublishPlaceId { get; set; }
    public string No { get; set; } = string.Empty;
    public string Period { get; set; } = string.Empty;
    public int LanguageId { get; set; }
    public int SubjectId { get; set; }

    public Language? Language { get; set; }
    public Subject? Subject { get; set; }
    public City? City { get; set; }
    public PublicationType? PublicationType { get; set; }
}

File: src/Web.UI/src/app/models/publication.ts
export type Publication = {
  id: number;
  name: string;
  typeId: number;
  typeName?: string;
  concessionaire: string;
  responsibleDirector: string;
  editor: string;
  year: string;
  journalNo: string;
  publishDate: string;
  publishPlaceId: number;
  publishPlaceName?: string;
  no: string;
  period: string;
  languageId: number;
  languageName?: string;
  subjectId: number;
  subjectTitle?: string;
};

export type CreatePublicationRequest = {
  name: string;
  typeId: number;
  concessionaire: string;
  responsibleDirector: string;
  editor: string;
  year: string;
  journalNo: string;
  publishDate: string;
  publishPlaceId: number;
  no: string;
  period: string;
  languageId: number;
  subjectId: number;
};

export type UpdatePublicationRequest = CreatePublicationRequest & {
  id: number;
};

export type PublicationSearchRequest = {
  name?: string;
  typeId?: number;
  publishPlaceId?: number;
  languageId?: number;
  subjectId?: number;
  publishDate?: string;
};

export type PublicationSearchResponse = {
  id: number;
  name: string;
  typeName?: string;
  publishPlaceName?: string;
  publishDate: string;
  languageName?: string;
  subjectTitle?: string;
  journalNo: string;
  no: string;
};


